// game/app_server/controllers/list.js
// JavaScript source code
/* GET 'home' page */
module.exports.infoSite = function (req, res) {
    res.render('list-display', {
        title: 'Recipes List',
        pageHeader: {
            title: 'Recipes'            
        },
    
        recipes: [{
            name: 'Waffles',
            description: 'Dessert made up of white flour and decoreated with whipped topping and fruits alongwith chocolate and maple syrup.',
            tomake1: 'Toast the frozen waffles for 1 minute in medium heat.',
            tomake2: 'Garnish it with whipped cream, fruits and syrups.'
            
        }, {
                name: 'Pancakes',
                description: 'Breakfast menu, that is made up of flour and eaten with maple syrup.',
                tomake1: 'Beat 2 eggs and mix it with 250 grams of flour.',
                tomake2: 'Add some baking powder to it and make thin cakes over heated pan. Turn them ocassionally.'
        }, {
                name: 'Cheesecake',
                description: 'Dessert made up of cream cheese and decorated with strawberries or other fruits of choice. It is set over gram crumbs.',
                tomake1: 'Thoroughly beat the cheese cream with sugar and some food colour.',
                tomake2: 'Set the cream over the base of crumbles grams and decorate it with fruits of your choice.'
        }]
    });
};
